SimpleNotePad.java

```diff
@@ -0,0 +1,222 @@
+        buildFileMenu();
+        buildEditMenu();
+        menuBar.add(fileMenu);
+        menuBar.add(editMenu);
+        setJMenuBar(menuBar);
+        add(new JScrollPane(userDataTextPane));
+        setPreferredSize(new Dimension(600,600));
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setVisible(true);
+        pack();
+    }
+
+    //create a File menu with open, print, save, new file and recent items in it
+    private void buildFileMenu(){
+        fileMenu.add(openMenuItem);
+        fileMenu.addSeparator();
+        fileMenu.add(printFileMenuItem);
+        fileMenu.addSeparator();
+        fileMenu.add(saveFileMenuItem);
+        fileMenu.addSeparator();
+        fileMenu.add(newFileMenuItem);
+        fileMenu.addSeparator();
+        fileMenu.add(recentMenuItem);
+
+        newFileMenuItem.addActionListener(this);
+        newFileMenuItem.setActionCommand("new");
+        saveFileMenuItem.addActionListener(this);
+        saveFileMenuItem.setActionCommand("save");
+        printFileMenuItem.addActionListener(this);
+        printFileMenuItem.setActionCommand("print");
+        recentMenuItem.addActionListener(this);
+        recentMenuItem.setActionCommand("recent");
+        openMenuItem.addActionListener(this);
+        openMenuItem.setActionCommand("open");
+    }
+
+    // create Edit menu with copy, paste and replace items in it
+    private void buildEditMenu(){
+        editMenu.add(copyMenuItem);
+        editMenu.add(pasteMenuItem);
+        editMenu.add(replaceMenuItem);
+
+        copyMenuItem.addActionListener(this);
+        copyMenuItem.setActionCommand("copy");
+        pasteMenuItem.addActionListener(this);
+        pasteMenuItem.setActionCommand("paste");
+        replaceMenuItem.addActionListener(this);
+        replaceMenuItem.setActionCommand("replace");
+    }
+
+    // build a list of recent open items
+    private void buildRecentMenu(){
+        recentMenuItem.removeAll();    //reset the list every time the method is called
+        for (int i = 0; i < recentlyOpenedFiles.length; i++) {
+            if (recentlyOpenedFiles[i] != null) {
+                JMenuItem recentFileMenuItem = new JMenuItem(recentlyOpenedFiles[i].getName());
+                recentFileMenuItem.addActionListener(this);
+                recentFileMenuItem.setActionCommand("recentFile" + i);
+                recentMenuItem.add(recentFileMenuItem);
+            }
+        }
+    }
+
+    private void addFileToRecentlyUsedFilesArray(File newFile) {
+        for (int i = 0; i < recentlyOpenedFiles.length-1; i++) {
+            // move the most recently opened item to next item
+            recentlyOpenedFiles[recentlyOpenedFiles.length-1-i] = recentlyOpenedFiles[recentlyOpenedFiles.length-2-i];
+        }
+        recentlyOpenedFiles[0] = newFile;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        int index = 0;
+        switch (e.getActionCommand()) {
```

Comment:
Should remove switch statement, don't use switch statement for object behavior

You can move each switch scenario into their own methods and then pass those methods/functions into the addActionListener methods above